cmd/bosun/expr/funcs.go

```diff
@@ -595,6 +603,56 @@ func Merge(e *State, T miniprofiler.Timer, series ...*Results) (*Results, error)
 	return res, nil
 }
 
+func LeftJoin(e *State, T miniprofiler.Timer, keysCSV, columnsCSV string, rowData ...*Results) (*Results, error) {
+	res := &Results{}
+	dataWidth := len(rowData)
+	if dataWidth == 0 {
+		return res, fmt.Errorf("leftjoin requires at least one item to populate rows")
+	}
+	keyColumns := strings.Split(keysCSV, ",")
+	dataColumns := strings.Split(columnsCSV, ",")
+	keyWidth := len(keyColumns)
+	keyIndex := make(map[string]int)
+	for i, v := range keyColumns {
+		keyIndex[v] = i
+	}
+	if len(dataColumns) != dataWidth {
+		return res, fmt.Errorf("mismatch in length of data rows and data labels")
+	}
+	t := Table{}
+	t.Columns = append(keyColumns, dataColumns...)
+	rowWidth := len(dataColumns) + len(keyColumns)
+	rowGroups := []opentsdb.TagSet{}
+	for i, r := range rowData {
+		if i == 0 {
+			for _, val := range r.Results {
+				row := make([]interface{}, rowWidth)
```

Comment:
There are string values since some of the columns are tags. If there is more complicated use of tables in the future I could add another field to the table type that specifies the type of each column